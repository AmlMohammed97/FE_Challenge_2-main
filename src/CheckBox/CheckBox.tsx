import './CheckBox.css';

import React from 'react';

export type Option = {
  label: string;
  value: string;
};

/**
 *
 *
 * @param {string} label - the label text of this component
 * @param {string} dataTestId - data test id to select it in unit tests
 * @param {boolean} isChecked - is this component checked
 * @param {Function} onChange - when checked options are changed,
 *                             they should be passed to outside
 */
type Props = {
  label: string;
  isChecked: boolean;
  onChange: () => void;
  dataTestId?: string;
};

const MultiCheck: React.FunctionComponent<Props> = ({ label, isChecked, onChange, dataTestId = 'checkbox' }): JSX.Element => {
  return (
    <div>
      <input type="checkbox" data-testid={dataTestId} checked={isChecked} onChange={onChange} />
      <label>{label}</label>
    </div>
  );
};

export default MultiCheck;
