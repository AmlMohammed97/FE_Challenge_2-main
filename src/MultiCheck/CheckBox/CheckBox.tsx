import './CheckBox.css';

import React from 'react';

export type Option = {
  label: string;
  value: string;
};

/**
 *
 *
 * @param {string} label - the label text of this component
 * @param {boolean} isChecked - is this component checked
 * @param {Function} onChange - when checked options are changed,
 *                             they should be passed to outside
 */
type Props = {
  label: string;
  isChecked: boolean;
  onChange: () => void;
};

const MultiCheck: React.FunctionComponent<Props> = ({ label, isChecked, onChange }): JSX.Element => {
  return (
    <div>
      <input type="checkbox" data-testid="checkbox" checked={isChecked} onChange={onChange} />
      <label>{label}</label>
    </div>
  );
};

export default MultiCheck;
