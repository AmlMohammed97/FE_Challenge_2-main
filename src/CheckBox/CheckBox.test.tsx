import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import CheckBox from '.';

describe('CheckBox', () => {
  describe('snapshot', () => {
    it('renders correctly', () => {
      const multiCheck = renderer.create(<CheckBox label="test checkbox label" onChange={() => {}} isChecked={true} />).toJSON();
      expect(multiCheck).toMatchSnapshot();
    });
  });
  it('renders the label', () => {
    render(<CheckBox label="test checkbox label" onChange={() => {}} isChecked={false} />);
    const label = screen.getByText('test checkbox label');
    expect(label).toBeInTheDocument();
  });
  it('renders un-checked', () => {
    render(<CheckBox label="" onChange={() => {}} isChecked={false} />);
    const checkbox = screen.getByTestId('checkbox');
    expect(checkbox).not.toBeChecked();
  });
  it('renders checked', () => {
    render(<CheckBox label="" onChange={() => {}} isChecked={true} />);
    const checkbox = screen.getByTestId('checkbox');
    expect(checkbox).toBeChecked();
  });
});
