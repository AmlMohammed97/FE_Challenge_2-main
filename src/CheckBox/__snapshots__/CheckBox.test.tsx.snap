// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox snapshot renders correctly 1`] = `
<div>
  <input
    checked={true}
    data-testid="checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <label>
    test checkbox label
  </label>
</div>
`;
